### PR TITLE
To add Stochastic Gradient Descent to Documentation

### PR DESCRIPTION
**Pull request by iramazanli**
It has been discussed before that adding description of Optimization algorithms to PyTorch Core documentation may result in a nice Optimization research tutorial. In the following tracking issue we mentioned about all the necessary algorithms and links to the originally published paper  https://github.com/pytorch/pytorch/issues/63236.

In this PR we are adding description of Stochastic Gradient Descent to the documentation.  

SGDalgo

### CODE REVIEW

File: torch/optim/sgd.py [hunk 1]
```diff
@@ -6,6 +6,32 @@
...
+            &\rule{110mm}{0.4pt}                                                                 \\
+            &\textbf{input}      : \gamma \text{ (lr)}, \: \theta_0 \text{ (params)}, \: f(\theta)
+                \text{ (objective)}, \: \lambda \text{ (weight decay)},                          \\
+            &\hspace{13mm} \:\mu \text{ (momentum)}, \:\tau \text{ (dampening)},\:nesterov\\[-1.ex]
+            &\rule{110mm}{0.4pt}                                                                 \\
+            &\textbf{for} \: t=1 \: \textbf{to} \: \ldots \: \textbf{do}                         \\
+            &\hspace{5mm}g_t           \leftarrow   \nabla_{\theta} f_t (\theta_{t-1})           \\
+            &\hspace{5mm}\textbf{if} \: \lambda \neq 0                                           \\
+            &\hspace{10mm} g_t \leftarrow g_t + \lambda  \theta_{t-1}                            \\
+            &\hspace{5mm}\textbf{if} \: t = 1                                                    \\
```
[the remark this answers is not in the document]
Comment (iramazanli) [reply torch/optim/sgd.py:20]: actually the `if t>1` condition happens inside `\mu \neq 0` condition.  however `t==1` condition is in general scope above, so it might be tricky to replace it here.

File: torch/optim/sgd.py [hunk 2]
```diff
@@ -6,6 +6,32 @@
...
+            &\hspace{10mm} \textbf{b}_t \leftarrow g_t                                           \\
+            &\hspace{5mm}\textbf{if} \: \mu \neq 0                                               \\
+            &\hspace{10mm}\textbf{if} \: t > 1                                                   \\
+            &\hspace{15mm} \textbf{b}_t \leftarrow \mu \textbf{b}_{t-1} + (1-\tau) g_t           \\
+            &\hspace{10mm}\textbf{if} \: nesterov                                                \\
+            &\hspace{15mm} g_t \leftarrow \mu g_{t-1} + (1-\tau) \textbf{b}_t                    \\
```
[the remark this answers is not in the document]
Comment (iramazanli) [reply torch/optim/sgd.py:26]: yes, i guess formulas get confused when i was trying to make a shortcut expression then return back to the plain version.

**Review by albanD — APPROVED** [review]
LGTM
